src/js/game/LevelMVC/LevelModel.js

```diff
@@ -666,7 +667,7 @@ module.exports = class LevelModel {
         // there is no entity
         && (frontEntity === undefined)
         // no lava or water
-        && (this.groundPlane.getBlockAt(position).blockType !== "water" && this.groundPlane.getBlockAt(position).blockType !== "lava");
+        && (entity === this.agent || (this.groundPlane.getBlockAt(position).blockType !== "water" && this.groundPlane.getBlockAt(position).blockType !== "lava"));
```

Comment:
I'd much prefer to see something like `entity.canWalkOnBlock(this.groundPlane.getBlockAt(position))`, and have `player` and `agent` simply define the method themselves

Reply:
Roger that 👍 

I can do that first thing tomorrow.